agiverse/agent/memory/embedding.py

```diff
@@ -0,0 +1,29 @@
+from typing import List, Optional
+import numpy as np
+import litellm
+from dotenv import load_dotenv
+import logging
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+logger = logging.getLogger(__name__)
+
+load_dotenv()
+
+class EmbeddingGenerator:
+    def __init__(self, api_key: str, model: str = "text-embedding-3-small"):
```

Comment:
Let's use `agent.get_model()` to manage all models in one place so users can conveniently get/set all of them